### PR TITLE
Authentication/SOAP: Fix SOAP Auth.

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Mantis Issue: https://mantis.ilias.de/view.php?id=24422

### CODE REVIEW

**Comment by mjansenDatabay** [comment]
@smeyer-ilias Any objections? The SOAP auth. is currently available/configurable for administrators but completely broken.

**Comment by mjansenDatabay** [comment]
Hi @smeyer-ilias  , the TB regularly monitors the list of PRs for old ones. Coud you please merge this or request feedback if you have any objections.

**Comment by smeyer-ilias** [comment]
Thanks for the PR. I will try to check/test the changes until Sep 18th.

File: Services/Authentication/classes/Frontend/class.ilAuthFrontendCredentialsSoap.php
Comment (smeyer-ilias) [inline Services/Authentication/classes/Frontend/class.ilAuthFrontendCredentialsSoap.php]: "??" requires PHP 7

File: Services/Init/classes/class.ilInitialisation.php
Comment (smeyer-ilias) [inline Services/Init/classes/class.ilInitialisation.php:1676]: isset on method calls is not suported.

File: Services/Init/classes/class.ilStartUpGUI.php
Comment (smeyer-ilias) [inline Services/Init/classes/class.ilStartUpGUI.php]: According to the constructor this should be $DIC->http()->request()

File: Services/Init/classes/class.ilStartUpGUI.php
Comment (smeyer-ilias) [inline Services/Init/classes/class.ilStartUpGUI.php]: PHP >= 7

**Review by smeyer-ilias — CHANGES_REQUESTED** [review]

**Comment by smeyer-ilias** [comment]
Sorry, but I cannot merge this due to issues for PHP 5.6 support.

File: Services/Init/classes/class.ilInitialisation.php
Comment (mjansenDatabay) [reply Services/Init/classes/class.ilInitialisation.php:1676]: This is not a method call, but array dereferencing.

**Review by mjansenDatabay — COMMENTED** [review]
Thx @smeyer-ilias ! I fixed all issues.

File: Services/Init/classes/class.ilInitialisation.php
Comment (smeyer-ilias) [reply Services/Init/classes/class.ilInitialisation.php:1676]: Correct. But fails with fatal: PHP Fatal error:  Cannot use isset() on the result of a function call (you can use "null !== func()" instead) in 5.6

**Comment by kergomard** [comment]
At yesterdays TB-Meeting we stumbled over this PR that has been around for a while: @smeyer-ilias could you please have a look at this again and merge if you agree. @mjansenDatabay has fixed the issues with PHP 5.6 and it would be a pity if we would run into merge errors, because we leave this too long.

**Comment by klees** [comment]
Hi @smeyer-ilias,

this came up again in our last TB meeting on the 12-07. Is there anything new regarding this PR? Is there anything we could do to speed up the process?

Best regards!

**Review by smeyer-ilias — APPROVED** [review]
Thanks. Merged to >= 5.4

**Comment by smeyer-ilias** [comment]
Already merged into >= 5.4

**Pull request closed without merging**
